tests/jobs.py

```diff
@@ -901,80 +901,6 @@ def do_schedule(function, function2):
         do_schedule()
         self.assertEquals(2, len(executor.queued_tasks))
 
-    def test_retry_still_in_executor(self):
```

Comment:
I spent a good chunk of time debugging this test failure, it fails because of this new logic in task execution:
```
            .join(DagRun, and_(TI.dag_id == DagRun.dag_id,
                               TI.execution_date == DagRun.execution_date,
                               DagRun.state == State.RUNNING,
                               DagRun.run_id.like(DagRun.ID_PREFIX + '%')))
            .distinct()
```

This logic makes sense (don't schedule task instances for dag runs that have failed).

In the test the dag run was getting set to failed (due to the logic below):
```
            elif unfinished_tasks and none_depends_on_past and no_dependencies_met:
                logging.info(
                    'Deadlock; marking run {} failed'.format(self))
                self.state = State.FAILED
```

and after this the scheduler was still running the task instances even though the dagrun failed.

This seems to be incorrect behavior. Maybe the dagrun should never have been set to failed in the first place though, is that what you mean?